### DISCUSSION TITLE
Display Language Selection

### DISCUSSION THREAD

**Discussion by galatival** [tethysplatform/tethys#634] [Ideas]
I want to be able to implement a way to change the display language translation in my tethys app, is there a recommended method to accomplish this? Thank you!

**Comment by swainn** [comment]
@galatival Tethys currently doesn't support this feature, however, we would like to. Tethys is built on Django and Django definitely has support for this. So it is just a matter of adding the feature to Tethys.

What languages are you interested in supporting? Would you be interested in helping with translation if this feature is added?

@msouff, one of my colleagues is interested in working on this, so I'm looping him in.

**Reply by ZoloKiala** [reply]
Dear swainn,

I can help with french.

Regards

Serge

**Reply by msouff** [reply]
Great. In theory I already have it set up and I've just been slowly working on the Spanish translations. Since @ZoloKiala wants to help with French what I'll do is run what I have so far to make sure it is working and push a working branch.

**Reply by galatival** [reply]
Thank you so much, I was able to implement it with the Django method pretty easily! We are looking to support Spanish and Portuguese, but some of my colleagues will be doing the translations, not me, so I don't think I'd be able to help much. 

Thanks again!

**Comment by swainn** [comment]
Great! Thank you, everyone. @msouff go ahead and open a WIP Pull Request after you push your branch.

**Comment by msouff** [comment]
I added #673 over the weekend.

This is a good start, but talking to @swainn  we may want to expand this to allow developers to provide multiple translations for the customizable site content for cases when the Tethys Portal is meant to be accessed from different parts of the world.

Currently, the customizable site contents are saved to the tethys database on the native language of the Server. The rest of the translations are not saved to the database and are responsive to the the language of the browser accessing it so no problem with those.

There is a README file in the new tethys_locale directory that shows how to add new translations so anyone can start working on that independently.
